daemon/cdi_linux.go

```diff
@@ -0,0 +1,42 @@
+package daemon
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/pkg/capabilities"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func registerCdiDriver() error {
+	driver := &deviceDriver{
+		capset:     capabilities.Set{"cdi": struct{}{}},
+		updateSpec: injectCdiDevices,
+	}
+
+	registerDeviceDriver("cdi", driver)
+
+	return nil
+}
+
+// injectCdiDevices is a updateSpecHander for the CDI device driver
+func injectCdiDevices(s *specs.Spec, dev *deviceInstance) error {
+	return injectCDI(s, dev.req.DeviceIDs)
+}
+
+// injectCDI injects a CDI device into the OCI spec.
+func injectCDI(s *specs.Spec, cdiDeviceNames []string) error {
+	if len(cdiDeviceNames) == 0 {
+		return nil
+	}
+
+	if cdiRegistry == nil {
+		return fmt.Errorf("CDI registry not initialized")
+	}
+
+	_, err := cdiRegistry.InjectDevices(s, cdiDeviceNames...)
+	if err != nil {
+		return fmt.Errorf("CDI device injection failed: %w", err)
+	}
+
+	return nil
+}
```

Comment:
None of this looks Linux specific, and if and when we get Windows (or other OS?) support for CDI, I guess it would move into cdi.go unchanged. Can it be put there now, with only the shorter `registerCdiDriver` function platform-split? (I notice that `registerDeviceDriver` is Linux-only, so we'd need to fix that before we can have Windows CDI support as well).

Reply:
It isn't linux specific, but due to `registerDeviceDriver` being linux-only I elected to keep this linux specific. I'm happy to work on a follow-up to address this afterwards.

Note that CDI doesn't currently support Windows device injection and this will have to be added to that package to work as expected.

Reply:
To be clear, I was referring to the `injectCdiDevices` and `injectCDI` functions. The main thing we get from having those in the non-platform-specific file is that `cdiRegistry.InjectDevices` is visibly not platform-limited to the next person to touch this code, and reduced refactoring impact if and when we do get Windows support in CDI.

Reply:
OK. I'll take another pass at this to improve things.

Reply:
@TBBle thanks for the review here. I think the latest revision has resolved your concerns here.